From 74265493be3a4826199329d4e6ff8fc49a33850f Mon Sep 17 00:00:00 2001
From: Puna2608 <opna2608@protonmail.com>
Date: Fri, 6 Oct 2023 21:06:13 +0200
Subject: [PATCH] persistent-cache-cpp: Lenient exception test matching

PersistentStringCacheImpl::exceptions attempts to very precisely match a returned std::system_error exception text to an expectation.
On LLVM's libcxx, the message returned from it doesn't match the expectation.

This patch changes the matching to only care about the part that doesn't depend on such an implementation detail.
---
 .../persistent_string_cache_impl_test.cpp                  | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/tests/core/internal/persistent_string_cache_impl/persistent_string_cache_impl_test.cpp b/tests/core/internal/persistent_string_cache_impl/persistent_string_cache_impl_test.cpp
index f3eee59..fedd466 100644
--- a/tests/core/internal/persistent_string_cache_impl/persistent_string_cache_impl_test.cpp
+++ b/tests/core/internal/persistent_string_cache_impl/persistent_string_cache_impl_test.cpp
@@ -24,6 +24,7 @@
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wctor-dtor-privacy"
 #include <gtest/gtest.h>
+#include <gmock/gmock.h>
 #pragma GCC diagnostic pop
 
 #include <iostream>
@@ -1295,9 +1296,9 @@ TEST(PersistentStringCacheImpl, exceptions)
             }
             catch (system_error const& e)
             {
-                EXPECT_EQ("PersistentStringCache: check_version(): bad version: \"nan\" (cache_path: " + TEST_DB +
-                          "): Unknown error 666",
-                          e.what());
+                EXPECT_THAT(
+                    e.what(),
+                    ::testing::StartsWith("PersistentStringCache: check_version(): bad version: \"nan\" (cache_path: " + TEST_DB + "): "));
             }
         }
 
-- 
2.40.1

